packages/headless/src/features/tab-set/tab-set-actions.ts

```diff
@@ -0,0 +1,40 @@
+import {RecordValue} from '@coveo/bueno';
+import {createAction} from '@reduxjs/toolkit';
+import {
+  requiredEmptyAllowedString,
+  requiredNonEmptyString,
+  validatePayload,
+} from '../../utils/validate-payload';
+
+export interface RegisterTabActionCreatorPayload {
+  /**
+   * A unique identifier for the tab
+   */
+  id: string;
+
+  /**
+   * The tab filter expression.
+   */
+  expression: string;
+}
+
+export const registerTab = createAction(
+  'tab/register',
+  (payload: RegisterTabActionCreatorPayload) => {
+    const schema = new RecordValue({
+      values: {
+        id: requiredNonEmptyString,
+        expression: requiredEmptyAllowedString,
```

Comment:
I think it would be a better API to accept `undefined expression` OR `empty expression` to both mean "no filter applied for this tab", versus only `empty expression` ?